rclcpp_action/test/test_client.cpp

```diff
@@ -1035,3 +1035,20 @@ TEST_F(TestClientAgainstServer, execute_rcl_errors)
       rclcpp::exceptions::RCLError);
   }
 }
+
+TEST_F(TestClientAgainstServer, test_configure_introspection)
+{
+  auto action_client = rclcpp_action::create_client<ActionType>(client_node, action_name);
+
+  EXPECT_THROW(
+    action_client->configure_introspection(
+      nullptr, rclcpp::SystemDefaultsQoS(), RCL_SERVICE_INTROSPECTION_CONTENTS),
+      std::invalid_argument);
+
+  EXPECT_NO_THROW(
+    action_client->configure_introspection(
+      client_node->get_clock(), rclcpp::SystemDefaultsQoS(), RCL_SERVICE_INTROSPECTION_CONTENTS));
+
+  // No method was found to make rcl_action_client_configure_internal_service_introspection return
+  // a value other than RCL_RET_OK. mocking_utils::patch_and_return does not work for this function.
```

Comment:
this is because `mocking_utils::patch_and_return` is not hot patch, right? it is impossible to patch `lib:rcl` functions in rclcpp test?

Reply:
No. The issue is related to the `rcl_publisher_options_t` parameter for rcl_action_client_configure_action_introspection().  
It will report build error as below while using mocking_utils::patch_and_return.
```
/root/ros2_src_rolling/src/ros2/rclcpp/rclcpp_action/test/mocking_utils/patch.hpp:143:3: error: no match for ‘operator==’ (operand types are ‘rcl_publisher_options_s’ and ‘rcl_publisher_options_s’)
  143 |   mmk_mock_define(
      |   ^~~~~~~~~~~~~~~
```